### PR TITLE
Add deprecation check for adjacency matrix setting

### PR DESCRIPTION
**Pull request by nik9000**
We dropped the adjacency matrix max clause setting in #46327 because we
believe that it is no longer required. From here on out you can use the
regular boolean max clause setting.

### CODE REVIEW

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java [hunk 1]
```diff
@@ -365,6 +365,14 @@ static DeprecationIssue checkIndexRoutingExcludeSetting(IndexMetadata indexMetad
         );
     }
 
+    static DeprecationIssue checkIndexMatrixFiltersSetting(IndexMetadata indexMetadata) {
+        return checkRemovedSetting(indexMetadata.getSettings(),
+            IndexSettings.MAX_ADJACENCY_MATRIX_FILTERS_SETTING,
+            "https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_settings_changes",
```
[the remark this answers is not in the document]
Comment (nik9000) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java:371]: I think that sentence is accurate from the perspective of 7.x. You can still specify the setting in 7.x and we'll respect it. But we won't in 8.0. Should it say "this setting will be ignored in 8.0. Use `indices.query.bool.max_clause_count` instead." Or something like that?

All of the links above are to the 8.0 breaking changes. Are they also wrong?

[the remark this answers is not in the document]
Comment (masseyke) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java:371]: The wording sounds good to me. I've fixed all the other links that were pointing to 8.0 documentation in the last week. We're actually now using shortened URLs that can be repointed to wherever the docs team wants them without a code change (https://ela.st). If you want to change this URL to "https://ela.st/es-deprecation-7-adjacency-matrix-filters-setting" and let the docs team know where to point that, that would work.

**Comment by nik9000** [comment]
@masseyke I've updated the message and url.

**Review by masseyke — APPROVED** [review]

**Comment by nik9000** [comment]
@elasticmachine update branch

**Pull request merged**